packages/salesforcedx-vscode-core/src/commands/forceSourceDeploySourcePath.ts

```diff
@@ -123,14 +126,17 @@ export class LibraryDeploySourcePathExecutor extends LibraryCommandletExecutor<
         throw new Error('SourceClient is not established');
       }
 
-      this.sourceClient.tooling.deployWithPaths = this.deployWrapper(
-        this.sourceClient.tooling.deployWithPaths
+      this.sourceClient.tooling.deploy = this.deployWrapper(
+        this.sourceClient.tooling.deploy
       );
 
-      await this.sourceClient.tooling.deployWithPaths({
-        paths: [response.data]
-      });
-      this.logMetric();
+      const registryAccess = new RegistryAccess();
+      const components = registryAccess.getComponentsFromPath(response.data);
+      const deployPromise = this.sourceClient.tooling.deploy({ components });
```

Comment:
What are we gaining by doing this instead of `await this.sourceClient.tooling.deploy(...` ?

Reply:
I suppose it's a minuscule benefit but it's to kick off the request before building the telemetry data.